From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Wed, 5 Aug 2026 11:27:23 +0200
Subject: [PATCH 38/67] Prevent errors from DownloadPrompter

Let embedders disable toolkit download prompts while keeping the secure default.

Sailfish supplies download lifecycle UI outside Gecko browser chrome.
---
 toolkit/components/downloads/DownloadUIHelper.sys.mjs | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/toolkit/components/downloads/DownloadUIHelper.sys.mjs b/toolkit/components/downloads/DownloadUIHelper.sys.mjs
index 3d8f80c5a8b6..5b9081cda305 100644
--- a/toolkit/components/downloads/DownloadUIHelper.sys.mjs
+++ b/toolkit/components/downloads/DownloadUIHelper.sys.mjs
@@ -146,7 +146,12 @@ export var DownloadUIHelper = {
  *        attach prompts to the most recently active window.
  */
 var DownloadPrompter = function (aParent) {
-  this._prompter = Services.ww.getNewPrompter(aParent);
+  this._prompter = Services.prefs.getBoolPref(
+    "browser.download.useToolkitPrompter",
+    true
+  )
+    ? Services.ww.getNewPrompter(aParent)
+    : null;
 };
 
 DownloadPrompter.prototype = {
